hbw/app/javascript/packs/hbw/components/shared/context/store.jsx

```diff
@@ -47,14 +47,18 @@ const withStoreContext = (WrappedComponent) => {
       this.getTaskById(taskId, cacheKey);
     };
 
-    onAssignment = async (taskId, cacheKey) => {
+    onAssignment = async (taskId, cacheKey, isAssigned) => {
       const { translator: t } = this.props.env;
       const { name, assignee } = await this.getTaskById(taskId, cacheKey);
 
       if (assignee !== null) {
-        Application.messenger.notice(t('notifications.new_assigned_task', {
-          task_name: name
-        }));
+        if (isAssigned) {
+          Application.messenger.notice(t('notifications.new_assigned_task', {
+            task_name: name
+          }));
+        } else {
+          this.onComplete(taskId);
```

Comment:
Semantically wrong — this is not completed task in fact.
Move code for task removing to separate method and use it in the `onComplete` and here.